GVRf/Framework/src/org/gearvrf/plugins/widget/GVRWidgetPluginActivity.java

```diff
@@ -0,0 +1,757 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.plugins.widget;
+
+import java.lang.reflect.Method;
+import java.util.Objects;
+
+import javax.microedition.khronos.egl.EGLContext;
+
+import org.gearvrf.GVRActivity;
+import org.gearvrf.GVRSceneObject;
+import org.gearvrf.GVRScript;
+import org.gearvrf.plugins.GVRPlugin;
+import org.gearvrf.scene_objects.GVRWidgetSceneObject;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.content.res.Configuration;
+import android.opengl.GLSurfaceView;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.Debug;
+import android.os.Handler;
+import android.util.Log;
+import android.view.Gravity;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.Window;
+import android.view.WindowManager;
+import android.widget.FrameLayout;
+
+import com.badlogic.gdx.ApplicationListener;
+import com.badlogic.gdx.Audio;
+import com.badlogic.gdx.Files;
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.Graphics;
+import com.badlogic.gdx.LifecycleListener;
+import com.badlogic.gdx.Net;
+import com.badlogic.gdx.Preferences;
+import com.badlogic.gdx.Application.ApplicationType;
+import com.badlogic.gdx.backends.android.AndroidApplication;
+import com.badlogic.gdx.backends.android.AndroidApplicationBase;
+import com.badlogic.gdx.backends.android.AndroidApplicationConfiguration;
+import com.badlogic.gdx.backends.android.AndroidAudio;
+import com.badlogic.gdx.backends.android.AndroidClipboard;
+import com.badlogic.gdx.backends.android.AndroidEventListener;
+import com.badlogic.gdx.backends.android.AndroidFiles;
+import com.badlogic.gdx.backends.android.AndroidGraphics;
+import com.badlogic.gdx.backends.android.AndroidInput;
+import com.badlogic.gdx.backends.android.AndroidInputFactory;
+import com.badlogic.gdx.backends.android.AndroidNet;
+import com.badlogic.gdx.backends.android.AndroidPreferences;
+import com.badlogic.gdx.backends.android.surfaceview.FillResolutionStrategy;
+import com.badlogic.gdx.utils.Array;
+import com.badlogic.gdx.utils.Clipboard;
+import com.badlogic.gdx.utils.GdxNativesLoader;
+
+/**
+ * This provides GVR (libGDX) widget lifecycle and context management and brings
+ * it together with GVR activity context. Base activity for GVR apps which use
+ * GVRWidgets
+ */
+public class GVRWidgetPluginActivity extends GVRActivity implements
```

Comment:
yes, since Libgdx and its context management happens through its own activity, so this activity serves purpose for creating the libgdx widget inside GVRF.